src/Parse/Yaml.php

```diff
@@ -19,8 +21,12 @@ class Yaml
      */
     public function parse($contents)
     {
-        $yaml = new Parser;
-        return $yaml->parse($contents);
+        try {
+            $yaml = new Parser;
+            return $yaml->parse($contents);
+        } catch (\Exception $e) {
+            throw new ParseException("A syntax error was detected in $fileName. " . $e->getMessage(), __LINE__, __FILE__);
```

Comment:
$fileName doesn't exist in this context. When testing it please make sure that an error in parsing a file gets handled by the exception handler in parseFile() and this handler doesn't just eat the exception